docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6324,380 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
 `p4.v1.TableEntry` message based on the value of architecture-specific table
 properties, but we may include on in future versions of the API.
 
+# GenericTable { #sec-generic-table}
+
+## GenericTable p4info { #sec-p4info-generic-table}
+
+`GenericTable` messages are used to map non-PSA P4-externs or non-P4 target-
+specific "fixed" features to their implementation in a generic way. The same can
+be achieved via `Extern`, but GenericTable provides a structured way in which
+every feature is represented as a set of match-fields and data-fields. The data
+consists of unions which are similar to actions. One use of GenericTable in a
+server backend is with TDI [Table Driven
+Interface](https://github.com/p4lang/tdi), but targets can use Generic table to
+map to their own specific feature implementations as well.  It defines the
+following fields
+
+* `generic_table_type_id`, an 8-bit unsigned integer which uniquely identifies
+  the generic_table_type in the context of the architecture.
+  This value is in the range of `[0x00, 0xff]`.
+  The ID in the preamble is created as `0x18 + generic_table_type_id + 16 bits
```

Comment:
If you mean something like the following, I suggest replacing the formula you wrote with the description below, since you seem to use `+` in two different meanings in the same formula, which can be very confusing.

The ID in the preamble is calculated as `(0x18 << 24) + (generic_table_type_id << 16) + (16-bit ID generated by the P4 compiler)`